### DISCUSSION TITLE
How to filter correctly with List

### DISCUSSION THREAD

**Discussion by MuslemRahimi** [pocketbase/pocketbase#2649] [Q&A] [answered]
Hi,

I want to lazy load elements from my posts collection. 

I want to give pocketbase a list of post id's that the user already have seen and want to extract only new posts where the id is not within the already seen post.

`
let output = await locals.pb.collection('posts').getList(data.startPage, data.endPage, {
    sort: '-created',
    filter: `!data.seenPostsId.some(seenId => seenId === id)`,
});`

Is there a way to do this in the above way?

**Comment by ganigeorgiev** [comment, marked as answer]
If you have an array with the seen post ids, you can construct a filter expression like `id != "id1" && id != "id2" ...`:

```js
const seenIds = ["id1", "id2", "id3"];

let output = await locals.pb.collection("posts").getList(data.startPage, data.endPage, {
    sort: '-created',
    filter: seenIds.map((id) => `id!="${id}"`).join("&&"),
})
```

**Reply by MuslemRahimi** [reply]
thank you so much
